src/Argument/ArgumentResolverTrait.php

```diff
@@ -69,11 +80,16 @@ public function reflectArguments(ReflectionFunctionAbstract $method, array $args
             }
 
             if ($class !== null) {
-                return $class->getName();
+                if ($param->isDefaultValueAvailable()) {
+                    return new ClassNameWithOptionalValue($class->getName(), $param->getDefaultValue());
+                }
+
+                return new ClassName($class->getName());
             }
 
             if ($param->isDefaultValueAvailable()) {
                 return $param->getDefaultValue();
+                //return new RawArgument($param->getDefaultValue());
```

Comment:
I wonder if this should actually be a RawArgument, don't think behaviour should be that it should resolve a string class name for example, thoughts?

Reply:
emm yeah. it's been some time, i'm going to need to run the tests and see what happens. i commented this for some reason but can't remember why.